pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -193,16 +195,79 @@ func (c *Cluster) Start() error {
 	}
 	logger.Infof("wait timeout for healthy OSDs during upgrade or restart is %q", c.clusterInfo.OsdUpgradeTimeout)
 
+	osdsToSkipReconcile, err := c.getOSDsToSkipReconcile()
+	if err != nil {
+		logger.Warningf("failed to get osds to skip reconcile. %v", err)
+	}
+
 	// prepare for updating existing OSDs
 	updateQueue, deployments, err := c.getOSDUpdateInfo(errs)
 	if err != nil {
 		return errors.Wrapf(err, "failed to get information about currently-running OSD Deployments in namespace %q", namespace)
 	}
-	osdsToSkipReconcile, err := c.getOSDsToSkipReconcile()
-	if err != nil {
-		logger.Warningf("failed to get osds to skip reconcile. %v", err)
+
+	var replaceConfig *OSDReplaceConfig
+	if c.spec.Storage.Store.UpdateStore == OSDStoreUpdateConfirmation {
+		logger.Infof("start migrating osds to backend store %q", c.spec.Storage.Store.Type)
+
+		replaceConfig, err = c.getOSDReplaceConfig()
+		if err != nil {
+			return errors.Wrapf(err, "failed to get OSD replace config")
+		}
+
+		// skip updating the OSDs that need migration. This is to prevent the reconciler from updating the backend store in the OSD deployments
+		// that have not been migrated to the new store yet.
+		for _, osd := range replaceConfig.OSDs {
+			updateQueue.Remove([]int{osd.ID})
+		}
+
+		if len(replaceConfig.OSDs) != 0 {
+			logger.Infof("total osds that need migration to new backend store %q : %d", c.spec.Storage.Store.Type, len(replaceConfig.OSDs))
+			pgHealthMsg, pgClean, err := cephclient.IsClusterClean(c.context, c.clusterInfo)
+			if err != nil {
+				return errors.Wrapf(err, "failed to check if the cluster is clean")
```

Comment:
If this is going to fail the reconcile, let's be clear about the reason.
```suggestion
				return errors.Wrapf(err, "before replacing OSDs, waiting for the PGs to be clean")
```